Fix python-bugzilla usage for setting assigned-on date.

ralphbean opened the pull request:
Fixes #668.

Cole renamed this in version 2.0.0.  That was quite a while ago, so let's just
push the minimum requirement up to that as a baseline and use the new method
name.

ralphbean committed:
Fix python-bugzilla usage for setting assigned-on date.

Fixes #668.

Cole renamed this in version 2.0.0.  That was quite a while ago, so let's just
push the minimum requirement up to that as a baseline and use the new method
name.

ssbarnea commented:
While this fixes the initial error, it still fail to do a pull on redhat.bugzilla.com:
```
ERROR:bugwarrior.services:Worker for [bz] failed: 'NoneType' object has no attribute 'timetuple'
Traceback (most recent call last):
File "/Users/ssbarnea/rdo/bugwarrior/bugwarrior/services/__init__.py", line 499, in _aggregate_issues
for issue in service.issues():
File "/Users/ssbarnea/rdo/bugwarrior/bugwarrior/services/bz.py", line 295, in issues
extra['assigned_on'] = self._get_assigned_date(issue)
File "/Users/ssbarnea/rdo/bugwarrior/bugwarrior/services/bz.py", line 316, in _get_assigned_date
assigned_date_datetime = datetime.datetime.fromtimestamp(time.mktime(assigned_date.timetuple()))
AttributeError: 'NoneType' object has no attribute 'timetuple'
```

ralphbean committed:
Fix logic in getting assigned date.

This used to assume that the last event that happened on the bug *must* be the
assignment event.  Of course, this isn't true.  It could have been assigned and
then changed in some other way.

ralphbean committed:
Style cleanup in the bugzilla service.

ralphbean commented:
Thanks @ssbarnea.  Fixing in a subsequent commit in a moment.
